docs/madr/decisions/010-timeout-policy.md

```diff
@@ -0,0 +1,251 @@
+
+* Status: accepted
+
+Technical Story: [#4739](https://github.com/kumahq/kuma/issues/4739)
+
+## Context and Problem Statement
+
+[New policy matching MADR](https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/005-policy-matching.md) introduces 
+a new approach how Kuma applies configuration to proxies. Rolling out strategy implies creating a new set of policies that 
+satisfy a new policy matching. Current MADR aims to define a timeout policy that will be compliant with 2.0 policies model.
+
+## Considered Options
+
+* Create `MeshUpstreamTimeout` policy
+* Create two separate policies: `MeshUpstreamTimeout` and `MeshDownstreamTimeout`
+* Create single `MeshTimeout` policy
+
+## Decision Outcome
+
+We will be crating single `MeshTimeout` policy that will cover both downstream and upstream timeouts.
+
+## Solution
+
+### How Timeout policy can be configured right now
+This is the default timeout configuration in Kuma right now:
+
+```yaml
+ conf:
+    connectTimeout: 5s # all protocols
+    tcp: 
+      idleTimeout: 1h 
+    http:
+      requestTimeout: 15s 
+      idleTimeout: 1h
+      streamIdleTimeout: 30m
+      maxStreamDuration: 0s
+```
+
+Why idle timeout can be configured on both TCP and HTTP connections?
+Is there a chance that client would like to configure different idle timeouts for database/Kafka and for HTTP services? - I can't come up on any valid case when this can be used.
+
+### What timeouts can we configure in envoy
+1. TCP specific
+    1. connect timeout
+    2. idle timeout
+2. HTTP/GRPC connections specific
+    1. connection timeouts
+        1. idle timeout
+        2. max connection duration
+    2. stream timeouts
+        1. request timeout
+        2. request headers timeout
+        3. stream idle timeout
+        4. max stream duration
+    3. route timeout
+        1. idle timeout
+        2. timeout
+        3. per try timeout
+        4. per try idle timeout
+        5. max stream duration
+           Detailed explanation of those timeouts can be found in [Envoy docs](https://www.envoyproxy.io/docs/envoy/latest/faq/configuration/timeouts).
+
+#### Common timeouts
+As we can see, `connection` and `idle timeout` can be configured for every connection type. There are two options here:
+
+1. Keep separate idle timeouts for TCP and HTTP
+2. Merge those two timeouts into single configuration
+
+Ad 1. It gives clients more fine-tuning of timeouts, but it can cause chaos
+Ad 2. Simpler config and you can still achieve the same result as previously by configuring it for `MeshSubset|MeshService|MeshServiceSubset`
+
+#### Route timeouts
+As for the route timeouts, we only care about `idle timeout` and `timeout` since `per try timeout` and `per try idle timeout` 
+are retry related timeouts and should be configured in retry policy.
+
+#### Request headers timeout
+Last one specific timeout is `request headers timeout` but this timeout makes no sense on outbound connections. 
```

Comment:
Well... you can have a misbehaving http client in the app and the "mesh operators" wanting to minimise the damage by setting this timeout but that's not comment. I wouldn't rule out this for outbound completely but I agree that would be really rare.